src/adapters/defi.ts

```diff
@@ -124,24 +125,22 @@ class Defi {
   }
 
   public async discordWalletWithdraw(body: string, msg: Message) {
-    const resp = await fetch(`${API_BASE_URL}/defi/withdraw`, {
+    const json = await this.jsonFetch(`${API_BASE_URL}/defi/withdraw`, {
       method: "POST",
       headers: { "Content-Type": "application/json" },
       body,
     })
-
-    const json = await resp.json()
-    if (json.error.includes("balance is not enough")) {
-      throw new InsufficientBalanceError({
-        discordId: msg.author.id,
-        message: msg,
-        errorMsg: "Your balance is not enough to complete the transaction",
-      })
-    }
-    if (json.error !== undefined) {
+    if (json.ok === false) {
+      if (json.error.includes("balance is not enough")) {
+        throw new InsufficientBalanceError({
+          discordId: msg.author.id,
+          message: msg,
+          errorMsg: "Your balance is not enough to complete the transaction",
+        })
+      }
       throw new Error(json.error)
     }
-    return json
+    return json.data
```

Comment:
```suggestion
    return json
```
không nên assume là bên chỗ gọi cần lấy `data` nên cứ trả về full object rồi để bên kia handle là được
note: nhớ sửa bên chỗ gọi method này nữa nhé